rust/xaynet-analytics/src/database/screen_route/repo.rs

```diff
@@ -1,50 +1,46 @@
-use anyhow::{Error, Result};
+use anyhow::{anyhow, Error, Result};
+use std::convert::{TryFrom, TryInto};
 
 use crate::database::{
     common::{IsarAdapter, Repo},
     isar::IsarDb,
-    screen_route::data_model::ScreenRoute,
+    screen_route::{adapter::ScreenRouteAdapter, data_model::ScreenRoute},
 };
 
-pub struct ScreenRouteRepo<'db> {
-    collection_name: &'db str,
-}
-
-impl<'db> ScreenRouteRepo<'db> {
-    pub fn new(collection_name: &'db str) -> Self {
-        Self { collection_name }
-    }
-}
-
-impl<'db> Repo<ScreenRoute> for ScreenRouteRepo<'db> {
-    fn add(&self, route: &ScreenRoute, db: &IsarDb) -> Result<(), Error> {
-        let mut object_builder = db.get_object_builder(&self.collection_name)?;
-        route.write_with_object_builder(&mut object_builder);
-        let object_id = db.get_object_id_from_str(&self.collection_name, &route.name)?;
+impl<'db> Repo<'db, ScreenRoute> for ScreenRoute {
```

Comment:
yes, all Repo trait impl except `get` and `get_all` for `AnalyticsEvent` are basically the same, so they can just become default implementations. `AnalyticsEvent` is different because it stores `ScreenRoute` within its model, so it has this extra `RelationalAdapter` layer

Reply:
of course!